deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -41,25 +45,47 @@ public abstract class OpenApiGeneratorCodeGenBase implements CodeGenProvider {
 
     private static final String DEFAULT_PACKAGE = "org.openapi.quarkus";
 
+    /**
+     * The input base directory from
+     *
+     * <pre>
+     * src/main
+     *
+     * <pre>
+     * directory.
+     * Ignored if INPUT_BASE_DIR is specified.
+     **/
     @Override
     public String inputDirectory() {
         return "openapi";
     }
 
+    @Override
+    public boolean shouldRun(Path sourceDir, Config config) {
+        String inputBaseDir = getInputBaseDirRelativeToModule(sourceDir, config);
+        if (inputBaseDir != null && !Files.isDirectory(Path.of(inputBaseDir))) {
+            throw new OpenApiGeneratorException(String.format("Invalid path on %s: %s", INPUT_BASE_DIR, inputBaseDir));
+        }
+        return inputBaseDir != null || Files.isDirectory(sourceDir);
```

Comment:
@hbelmiro the original method, which is on CodeGenProvider, can return false, and it was returning false until I overwrote it in this class for adding the new functionality. The code, as it is now, works as it did before when inputBaseDir is null (returning Files.isDirectory(sourceDir)). If you want me to change it, I'm OK, just want you to know that it's not the way it's working right now

Reply:
I think we're not completely understanding each other. Let's clarify.

1) I understand that this method can return false. But in this case, we think that rather than returning false and simply not running (silently ignore), it would be clearer to the user of the extension if it threw an exception, which would make the application build fail.

2) I also understand that before your change we were silently returning false. Now your change revealed that and we can improve it.

>  If you want me to change it, I'm OK, just want you to know that it's not the way it's working right now.

I'm not sure I'm following you. Do you mean before your change it was returning false (item 2 above)?

Reply:
Understood, we're aligned now :) I made the change to throw an exception when the path (either default or custom) is not valid

Reply:
@hbelmiro now we have a problem... the tests fail (all of them) because the task generate-code-test tries to find the folder .../src/test/openapi and, as it doesn't exist, throws an exception... maybe that's why you were returning false as a default?

Reply:
Hmm... I don't think we should search in src/test/openapi.
But anyway, I think it's better to create an issue for this particular improvement, otherwise your feature will take too long to complete. And also you'll never want to contribute again because I'm asking you to do more things than you intended :)
So, can you please just revert it to your original version? I'll approve it.
And sorry for making you waste your time again.

Reply:
@ricardozanini FYI, I created https://github.com/quarkiverse/quarkus-openapi-generator/issues/192 to track this improvement since it's not as simple as I was thinking.

Reply:
hahaha all good @hbelmiro !! 
I understand why you wanted to change it, it seemed pretty straightforward... until it wasn't due to the tests. I'll try to have a look at it in the following days, but I do appreciate closing this feature to continue with our project and continue with that investigation later :D